lib/policies/proxy/proxy.js

```diff
@@ -12,73 +12,67 @@ const createStrategy = (strategy, proxyOptions, endpointUrls) => {
 module.exports = function (params, config) {
   const serviceEndpointKey = params.serviceEndpoint;
   const endpoint = config.gatewayConfig.serviceEndpoints[serviceEndpointKey];
-  const proxyOptions = parseProxyOptions(params.proxyOptions, endpoint);
 
-  if (!endpoint) {
-    throw new Error(
-      `service endpoint ${serviceEndpointKey} (referenced in 'proxy' ` +
-      'policy configuration) does not exist');
+  if (!endpoint) { // Note: one day this can be ensured by JSON Schema, when $data keyword will be avaiable.
+    throw new Error(`service endpoint ${serviceEndpointKey} (referenced in proxy policy configuration) does not exist`);
   }
 
-  if (!endpoint.url && !endpoint.urls) {
-    throw new Error(
-      `service endpoint ${serviceEndpointKey} (referenced in 'proxy' ` +
-      'policy configuration) does not contain a `url` or `urls` property');
+  const proxyOptions = parseProxyOptions(params.proxyOptions, endpoint);
+  const intermediateProxyUrl = process.env.HTTP_PROXY || params.proxyUrl;
```

Comment:
process.env.http_proxy is not the same as process.env.HTTP_PROXY
add support for both. 

if you are refactoring it anyway add no_proxy support

Reply:
Are we sure we want to support both?

You can define environment variables in any format you like, that's for sure. However I haven't been able to find any other meaningful project that's using environment variables in lower case format and, if we support it here, we either should support this scenario everywhere in the gateway (and it's a lot of places)  or we'd be inconsistent (like — some variables are ok to be lowercase, some are ok to be both, some should be just uppercase).

I think in this case is way better to be opinionated (it's uppercase, period) rather than trying to support every single use case. (camel case, kebab case, something_new case). Plus, uppercase it's _really_ the standard way.

@kevinswiber what's your thought here?

---

Let's postpone `no_proxy` for now — I do not know what's that about and I want to move forward with #647 and this is blocking me now.

Reply:
It turns out minikube is reasoning with such lowercase environment variable. So maybe it should stay lowercase? I really do not know…

![image](https://user-images.githubusercontent.com/1416224/37511724-2c663a7a-2900-11e8-8bef-56faac4443a7.png)

Reply:
https://unix.stackexchange.com/questions/212894/whats-the-right-format-for-the-http-proxy-environment-variable-caps-or-no-ca

So it really seems like nobody knows.

Reply:
Why not have both? Ubuntu supports both variants.

Reply:
Yeah — it probably makes sense — I just want to make sure everybody agrees. Let's discuss it with @altsang and @kevinswiber during the standup and we'll go from there